langchain/src/retrievers/parent_document.ts

```diff
@@ -159,7 +162,11 @@ export class ParentDocumentRetriever extends MultiVectorRetriever {
     for (let i = 0; i < parentDocs.length; i += 1) {
       const parentDoc = parentDocs[i];
       const parentDocId = parentDocIds[i];
-      const subDocs = await this.childSplitter.splitDocuments([parentDoc]);
+      const chunkHeaderOptions = chunkHeader && parentDoc?.metadata?.chunkHeader ? {
+        chunkHeader: parentDoc.metadata.chunkHeader,
```

Comment:
Depending on this to be in the `metadata` field of the document is a bit weird - could we make this a property on the class instead? Like `this.childSplitterOptions`?

Reply:
Ignore the above, I see the problem. Let me think about it a little longer - would be nice to not have a dependency on an untyped metadata key like this.

Reply:
I think we just have to the same way as when passing chunkHeader to parent chunk. It will be better solution. Thanks for spotting